File: src/main/java/com/gammon/pcms/service/HrService.java
package com.gammon.pcms.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.gammon.pcms.dao.hr.HrUserRepository;
import com.gammon.pcms.model.hr.HrUser;

@Service
@Transactional("hrTransactionManager")
public class HrService {

	@Autowired
	private HrUserRepository hrUserRepository;
	
	public HrUser findByUsername(String username){
		try{
			HrUser hrUser = hrUserRepository.findByUsername(username);
			return hrUser;
		} catch(Exception e){
			throw new NullPointerException();
		}
	}
	
	public HrUser findByEmployeeId(String employeeId){
		return hrUserRepository.findByEmployeeId(employeeId); 
	}
	
	public HrUser findByEmail(String email){
		return hrUserRepository.findByEmail(email);
	}

	public List<HrUser> findAll() {
		return hrUserRepository.findAll();
	}
	
	public List<HrUser> findByUsernameIsNotNull(){
		return hrUserRepository.findByUsernameIsNotNull();
	}

}

File: src/main/webapps/frontend-angular/js/service/hr-service.js
mainApp.service('hrService', ['$http', '$q', 'GlobalHelper', function($http, $q, GlobalHelper) {
	// Return public API.
    return({
    	findAll:	findAll,
    	findByEmail:	findByEmail,
    	findByEmployeeId:	findByEmployeeId,
    	findByUsername:	findByUsername,
    	findByUsernameIsNotNull: findByUsernameIsNotNull
    });
   
    function findAll(){
    	var request = $http.get('service/hr/findAll');
    	return( request.then( GlobalHelper.handleSuccess, GlobalHelper.handleError ) );
    }

    function findByEmail(){
    	var request = $http.get('service/hr/findByEmail');
    	return( request.then( GlobalHelper.handleSuccess, GlobalHelper.handleError ) );
    }

    function findByEmployeeId(){
    	var request = $http.get('service/hr/findByEmployeeId');
    	return( request.then( GlobalHelper.handleSuccess, GlobalHelper.handleError ) );
    }

    function findByUsername(){
    	var request = $http.get('service/hr/findByUsername');
    	return( request.then( GlobalHelper.handleSuccess, GlobalHelper.handleError ) );
    }

    function findByUsernameIsNotNull(){
    	var request = $http.get('service/hr/findByUsernameIsNotNull');
    	return( request.then( GlobalHelper.handleSuccess, GlobalHelper.handleError ) );
    }

}]);




